### PR TITLE
python: more linting

### PR DESCRIPTION
**Pull request by koivunej**
Ruff has "B" class of lints, including B018 which will nag on useless expressions, related to #4719. Enable such lints and fix the existing issues.

Most notably:
- https://beta.ruff.rs/docs/rules/mutable-argument-default/
- https://beta.ruff.rs/docs/rules/assert-false/

### CODE REVIEW

File: test_runner/fixtures/neon_fixtures.py [hunk 1]
```diff
@@ -2721,7 +2721,9 @@ def __init__(self, port: int, auth_token: Optional[str] = None):
     def check_status(self):
         self.get(f"http://localhost:{self.port}/v1/status").raise_for_status()
 
-    def debug_dump(self, params: Dict[str, str] = {}) -> Dict[str, Any]:
+    def debug_dump(self, params: Optional[Dict[str, str]] = None) -> Dict[str, Any]:
+        if params is None:
```
Comment (vadim2404) [inline test_runner/fixtures/neon_fixtures.py:2725]: NIT:

```python
params = params or {}
```

File: test_runner/performance/test_gc_feedback.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def test_gc_feedback(neon_env_builder: NeonEnvBuilder, zenbenchmark: NeonBenchma
         # without modifying the earlier parts of the table.
         for step in range(n_steps):
             cur.execute(f"INSERT INTO t (step) SELECT {step} FROM generate_series(1, {step_size})")
-            for i in range(n_update_iters):
+            for _i in range(n_update_iters):
```
Comment (vadim2404) [inline test_runner/performance/test_gc_feedback.py:50]: if it's unused, you can leave just `_` everywhere

File: test_runner/performance/test_parallel_copy_to.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 
 async def repeat_bytes(buf, repetitions: int):
-    for i in range(repetitions):
+    for _i in range(repetitions):
```
Comment (vadim2404) [inline test_runner/performance/test_parallel_copy_to.py:9]: Can it be replaced with 

```python
yield from itertools.repeat(buf, repetitions)
```

???

File: test_runner/regress/test_proxy.py [hunk 4]
```diff
@@ -229,7 +231,9 @@ def q(sql: str, params: List[Any] = []) -> Any:
 def test_sql_over_http_output_options(static_proxy: NeonProxy):
     static_proxy.safe_psql("create role http2 with login password 'http2' superuser")
 
-    def q(sql: str, raw_text: bool, array_mode: bool, params: List[Any] = []) -> Any:
+    def q(sql: str, raw_text: bool, array_mode: bool, params: Optional[List[Any]] = None) -> Any:
+        if params is None:
```
Comment (vadim2404) [inline test_runner/regress/test_proxy.py:235]:
```python
params = params or []
```

File: test_runner/regress/test_wal_acceptor.py [hunk 5]
```diff
@@ -46,8 +46,10 @@ def wait_lsn_force_checkpoint(
     timeline_id: TimelineId,
     endpoint: Endpoint,
     ps: NeonPageserver,
-    pageserver_conn_options={},
+    pageserver_conn_options = None,
 ):
+    if pageserver_conn_options is None:
```
Comment (vadim2404) [inline test_runner/regress/test_wal_acceptor.py:51]: again about liner

File: test_runner/performance/test_parallel_copy_to.py (hunk 3 above)
Comment (koivunej) [reply test_runner/performance/test_parallel_copy_to.py:9]: Another `ruff` autofix. but perhaps in this case we should.

**Review by bayandin — APPROVED** [review]

**Review by vadim2404 — APPROVED** [review]

File: pyproject.toml [hunk 6]
```diff
@@ -78,11 +78,13 @@ module = [
...
 [tool.ruff]
+target-version = "py39"
 extend-exclude = ["vendor/"]
 ignore = ["E501"]
 select = [
     "E", # pycodestyle
     "F", # Pyflakes
     "I", # isort
     "W", # pycodestyle
+    "B", # bugbear derived, useless expression
```
Comment (koivunej) [inline pyproject.toml:89]:
```suggestion
    "B", # bugbear
```

**Pull request merged**